Model/Api/CreateOrder.php

```diff
@@ -628,6 +628,11 @@ public function validateTax($quote, $transaction)
      */
     public function validateShippingCost($quote, $transaction)
     {
+        // Skip validation if Mirasvit_Credit is used.
+        // Rely on Model\Api\CreateOrder::validateTotalAmount which is called next.
+        if ($quote->getShippingAddress()->getCreditAmount()) {
+            return;
+        }
```

Comment:
This case was caught when manually tested compatibility with various internal and 3rd party modules (discounts, gift cards, rewards, store credit). Basically, Mirasvit_Credit does some unusual things with address totals that produces error in our validation method, but the order does not fail if skipped. Anyway, there's grand total validation called after this method that we can consider a fallback if this doesn't run.